File: decision_tree/r/dt.r
library(caret)
library(rpart)
library(rpart.plot)

# Load the dataset (Iris dataset as an example)
data(iris)

# Split the dataset into training and testing sets
set.seed(42)
train_indices <- createDataPartition(iris$Species, p = 0.8, list = FALSE)
train_data <- iris[train_indices, ]
test_data <- iris[-train_indices, ]

# Create a decision tree classifier
classifier <- rpart(Species ~ ., data = train_data)

# Visualize the decision tree
rpart.plot(classifier)

# Make predictions on the test data
y_pred <- predict(classifier, newdata = test_data, type = "class")

# Calculate the accuracy of the model
accuracy <- sum(y_pred == test_data$Species) / length(test_data$Species)
print(paste("Accuracy:", accuracy))

File: decision_tree/matlab/dt.m
% Load the dataset (Iris dataset as an example)
iris = readtable('iris.csv');
X = iris(:, 1:4);
y = iris(:, 5);

% Split the dataset into training and testing sets
rng('default'); % Set the random seed for reproducibility
[trainIndices, testIndices] = crossvalind('HoldOut', size(iris, 1), 0.2);
X_train = X(trainIndices, :);
y_train = y(trainIndices, :);
X_test = X(testIndices, :);
y_test = y(testIndices, :);

% Create a decision tree classifier
classifier = fitctree(X_train, y_train);

% Make predictions on the test data
y_pred = predict(classifier, X_test);

% Calculate the accuracy of the model
accuracy = sum(strcmp(y_pred, y_test)) / numel(y_test);
disp(['Accuracy: ', num2str(accuracy)]);
